[composite compliance] fix max_pool1d

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #84137
* __->__ #84127
* #84125

max_pool1d has a fast path for CPU tensors that do not require grad that
directly accesses the data_ptr. This PR makes the change that if the
input Tensor is a Tensor Subclass, then we want to walk through the
"slow path" of calling max_pool1d_with_indices.

Test Plan:
- wait for tests

zou3519 committed:
[composite compliance] fix max_pool1d

max_pool1d has a fast path for CPU tensors that do not require grad that
directly accesses the data_ptr. This PR makes the change that if the
input Tensor is a Tensor Subclass, then we want to walk through the
"slow path" of calling max_pool1d_with_indices.

Test Plan:
- wait for tests

[ghstack-poisoned]

zou3519 committed:
Update on "[composite compliance] fix max_pool1d"


max_pool1d has a fast path for CPU tensors that do not require grad that
directly accesses the data_ptr. This PR makes the change that if the
input Tensor is a Tensor Subclass, then we want to walk through the
"slow path" of calling max_pool1d_with_indices.

Test Plan:
- wait for tests

[ghstack-poisoned]

kshitij12345 reviewed (approved):
LGTM (given that the CI is green)! Thank you :)

zou3519 committed:
Update on "[composite compliance] fix max_pool1d"


max_pool1d has a fast path for CPU tensors that do not require grad that
directly accesses the data_ptr. This PR makes the change that if the
input Tensor is a Tensor Subclass, then we want to walk through the
"slow path" of calling max_pool1d_with_indices.

Test Plan:
- wait for tests

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge -f "flaky test in base PR"

zou3519 commented:
@pytorchbot merge -f "flaky test in base PR"
